build.gradle.kts

```diff
@@ -0,0 +1,268 @@
+
+    repositories {
+        maven(url = "https://repo.gradle.org/gradle/libs-releases")
+        maven(url = "https://repo.gradle.org/gradle/libs-milestones")
+        maven(url = "https://repo.gradle.org/gradle/libs-snapshots")
+    }
+
+    // patchExternalModules lives in the root project - we need to activate normalization there, too.
+    normalization {
+        runtimeClasspath {
+            ignore("org/gradle/build-receipt.properties")
+        }
+    }
+}
+
+apply(plugin = "gradlebuild.cleanup")
+apply(plugin = "gradlebuild.available-java-installations")
+apply(plugin = "gradlebuild.buildscan")
+apply(from = "gradle/versioning.gradle")
+apply(from = "gradle/dependencies.gradle")
+apply(plugin = "gradlebuild.minify")
+apply(from = "gradle/testDependencies.gradle")
+apply(plugin = "gradlebuild.wrapper")
+apply(plugin = "gradlebuild.ide")
+apply(plugin = "gradlebuild.no-resolution-at-configuration")
+apply(plugin = "gradlebuild.update-versions")
+apply(plugin = "gradlebuild.dependency-vulnerabilities")
+apply(plugin = "gradlebuild.add-verify-production-environment-task")
+
+
+subprojects {
+    version = rootProject.version
+
+    if (project in javaProjects) {
+        apply(plugin = "gradlebuild.java-projects")
+        apply(plugin = "gradlebuild.dependencies-metadata-rules")
+    }
+
+    if (project in publishedProjects) {
+        apply(plugin = "gradlebuild.publish-public-libraries")
+    }
+
+    apply(from = "$rootDir/gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts")
+    apply(plugin = "gradlebuild.task-properties-validation")
+    apply(plugin = "gradlebuild.test-files-cleanup")
+}
+
+configurations {
+    "coreRuntime" {
+        attributes { attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage::class.java, Usage.JAVA_RUNTIME)) }
+        isCanBeResolved = true
+        isCanBeConsumed = false
+        isVisible = false
+    }
+    "coreRuntimeExtensions" {
+        attributes { attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage::class.java, Usage.JAVA_RUNTIME)) }
+        isCanBeResolved = true
+        isCanBeConsumed = false
+        isVisible = false
+    }
+    "externalModules" {
+        isVisible = false
+    }
+    // Configuration used to resolve external modules before patching them with versions from core runtime
+    "externalModulesRuntime" {
+        isVisible = false
+        extendsFrom(configurations["coreRuntime"])
+        extendsFrom(configurations["externalModules"])
+    }
+    // Combines the 'coreRuntime' with the patched external module jars
+    "runtime" {
+        isVisible = false
+        extendsFrom(configurations["coreRuntime"])
+    }
+    "gradlePlugins" {
+        isVisible = false
+    }
+    "testRuntime" {
+        extendsFrom(configurations["runtime"])
+        extendsFrom(configurations["gradlePlugins"])
+    }
+
+    all {
+        attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage::class.java, Usage.JAVA_RUNTIME))
```

Comment:
:o: Doesn't this `attribute` declaration make previous `attribute` declarations unnecessary?